### PR TITLE
Add MOTP support

### PR DESCRIPTION
**Pull request by orange-elephant**
#608 Adds support for mOTP codes

I saw that this proposal had been accepted but no development had happened so I made this implementation. I am also happy to work on it further if changes are required :)

### CODE REVIEW

**Comment by alexbakker** [comment]
Thanks a lot for picking this up! I hope to have some time to look at it soon.

File: app/src/main/java/com/beemdevelopment/aegis/ui/EditEntryActivity.java [hunk 1]
```diff
@@ -679,7 +694,11 @@ private VaultEntry parseEntry() throws ParseException {
         byte[] secret;
         try {
             String secretString = new String(EditTextHelper.getEditTextChars(_textSecret));
-            secret = GoogleAuthInfo.parseSecret(secretString);
+            if (lowerCasedType.equals(MotpInfo.ID)) {
+                secret = Hex.decode(secretString);
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/ui/EditEntryActivity.java:698]: If this fails, a misleading "Secret is not valid base32" error dialog is shown.

File: app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java [hunk 2]
```diff
@@ -285,6 +288,8 @@ public Uri getUri() {
 
         if (_info instanceof YandexInfo) {
             builder.appendQueryParameter("pin", Base32.encode(((YandexInfo) _info).getPin()));
+        } else if (_info instanceof MotpInfo) {
+            builder.appendQueryParameter("pin", Base32.encode(((MotpInfo) _info).getPin()));
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java:292]: Is there a spec for the format of QR codes?. Is the PIN supposed to be included in the QR code? And if so, is the usage of base32 encoding correct here?

File: app/src/main/res/values/arrays.xml [hunk 3]
```diff
@@ -11,6 +11,7 @@
         <item>HOTP</item>
         <item>Steam</item>
         <item>Yandex</item>
+        <item>MOTP</item>
```
Comment (alexbakker) [inline app/src/main/res/values/arrays.xml:14]: What is the correct way to abbreviate Mobile-OTP? Is it mOTP or MOTP?

File: app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java [hunk 4]
```diff
@@ -89,12 +89,13 @@ public static GoogleAuthInfo parseUri(Uri uri) throws GoogleAuthInfoException {
                     info = hotpInfo;
                     break;
                 case YandexInfo.HOST_ID:
+                case MotpInfo.ID:
```
Comment (alexbakker) [inline app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java:92]: I'm a little confused to see that we're encoding mOTP secrets as hexadecimal everywhere, except for the QR codes. Is this the convention for mOTP?

**Review by alexbakker — CHANGES_REQUESTED** [review]
This is great work! I've done a preliminary review and added a couple of comments.

File: app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java (hunk 4 above)
Comment (orange-elephant) [reply app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java:92]: You're right the secrets are hex, will update this. my use case of pfSense wontwon't generate QR codes for MOTP whereas it does for google authenticator, so the only use case I tested for QR codes was Aegis to Aegis

File: app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java (hunk 2 above)
Comment (orange-elephant) [reply app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java:292]: The spec agreed upon by the creators of MOTP appears to be this https://www.token2.com/?content=motpqr as referenced here http://motp.sourceforge.net/#7 and Hex encoded secrets should definitely be used, will change this

File: app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java (hunk 4 above)
Comment (alexbakker) [reply app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java:92]: Good find. It also looks like the PIN should not be included in the URI, according to that spec.

File: app/src/main/res/values/arrays.xml (hunk 3 above)
Comment (orange-elephant) [reply app/src/main/res/values/arrays.xml:14]: I think it should be MOTP, however I've seen mOTP quite a lot so there doesn't seem to be great consensus :(

Comment (alexbakker) [reply app/src/main/res/values/arrays.xml:14]: Alright, let's leave it at "MOTP" and wait to see if someone complains about it.

**Comment by alexbakker** [comment]
Looks good! Could you squash everything into a single commit?

**Review by alexbakker — APPROVED** [review]
Thanks a lot!

**Pull request merged**